index.js

```diff
@@ -1,73 +1,90 @@
-const { Client, Collection, GatewayIntentBits } = require('discord.js');
-const { token, guildId } = require('./config.json');
-
-const fs = require('fs')
-const path = require("path")
+const { Client, Collection, GatewayIntentBits, Events } = require('discord.js');
+const { token } = require('./config.json');
+const fs = require('fs');
+const path = require('path');
 // Create a new client instance
-const client = new Client({ intents: [GatewayIntentBits.Guilds, GatewayIntentBits.GuildMessages, GatewayIntentBits.DirectMessages, GatewayIntentBits.GuildVoiceStates, GatewayIntentBits.GuildMembers] });
+const client = new Client({
+  intents: [
+    GatewayIntentBits.Guilds,
+    GatewayIntentBits.GuildMessages,
+    GatewayIntentBits.DirectMessages,
+    GatewayIntentBits.GuildVoiceStates,
+    GatewayIntentBits.GuildMembers,
+  ],
+});
 
 // Register Commands
 client.commands = new Collection();
-const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));
+const commandFiles = fs.readdirSync('./commands').filter((file) => file.endsWith('.js'));
 
 for (const file of commandFiles) {
-	const command = require(`./commands/${file}`);
-	client.commands.set(command.data.name, command);
+  const command = require(`./commands/${file}`);
+  client.commands.set(command.data.name, command);
 }
 
 
 
 client.once('ready', () => {
-	console.log('Ready!');
+  console.log('Ready!');
 });
 
-client.on('interactionCreate', async interaction => {
-	if (!interaction.isCommand()) return;
+client.on(Events.InteractionCreate, async (interaction) => {
+  try {
+    if (interaction.isButton()) {
+      const buttonUser = interaction.user.id;
+      const commandUser = interaction.message.interaction.user.id;
+      if (buttonUser === commandUser) {

```

Comment:
oh does this limit to only one person playing at one time?

Reply:
Since the message/interaction is visible to everyone in the channel, they will see the options to hit or stay. This will restrict the interaction to only be used by the person that initiated the interaction. This shouldn't prevent multiple players from playing at the same time though.

Reply:
Same with this. We can try and test this in my test server when you got some time!